protocol/linkpreview_unfurler_status.go

```diff
@@ -0,0 +1,194 @@
+	return &StatusUnfurler{
+		m:      messenger,
+		logger: logger,
+		url:    URL,
+	}
+}
+
+func buildThumbnail(image *images.IdentityImage, thumbnail *common.LinkPreviewThumbnail) error {
+	if image.IsEmpty() {
+		return nil
+	}
+
+	var err error
+
+	thumbnail.Width, thumbnail.Height, err = images.GetImageDimensions(image.Payload)
+	if err != nil {
+		return fmt.Errorf("failed to get image dimensions: %w", err)
+	}
+
+	thumbnail.DataURI, err = image.GetDataURI()
+	if err != nil {
+		return fmt.Errorf("failed to get data uri: %w", err)
+	}
+
+	return nil
+}
+
+func (u *StatusUnfurler) buildContactData(contactData *ContactURLData) (*common.StatusContactLinkPreview, error) {
+	c := new(common.StatusContactLinkPreview)
+	c.PublicKey = contactData.PublicKey
+	c.DisplayName = contactData.DisplayName
+	c.Description = contactData.Description
+
+	contactID, err := multiformat.DeserializeCompressedKey(contactData.PublicKey)
+	if err != nil {
+		return nil, err
+	}
+
+	contact := u.m.GetContactByID(contactID)
+	if contact == nil {
+		return c, nil
+	}
+
+	// TODO: Should we try to fetch from waku?
+	//if contact == nil {
+	//	if contact, err = u.m.RequestContactInfoFromMailserver(contactData.PublicKey, true); err != nil {
+	//		u.logger.Warn("StatusUnfurler: failed to request contact info from mailserver")
+	//		return c, err
+	//	}
+	//}
+
+	if image, ok := contact.Images[images.SmallDimName]; ok {
+		if err = buildThumbnail(&image, &c.Icon); err != nil {
+			return c, fmt.Errorf("failed to set thumbnail: %w", err)
+		}
+	}
+
+	return c, nil
+}
+
+func (u *StatusUnfurler) fillCommunityImages(community *communities.Community, icon *common.LinkPreviewThumbnail, banner *common.LinkPreviewThumbnail) error {
+	if image, ok := community.Images()[images.SmallDimName]; ok {
+		if err := buildThumbnail(&images.IdentityImage{Payload: image.Payload}, icon); err != nil {
+			u.logger.Warn("unfurling status link: failed to set community thumbnail", zap.Error(err))
+		}
+	}
+
+	if image, ok := community.Images()[images.BannerIdentityName]; ok {
+		if err := buildThumbnail(&images.IdentityImage{Payload: image.Payload}, banner); err != nil {
+			u.logger.Warn("unfurling status link: failed to set community banner", zap.Error(err))
+		}
+	}
+
+	return nil
+}
+
+func (u *StatusUnfurler) buildCommunityData(data *CommunityURLData) (*common.StatusCommunityLinkPreview, error) {
+	c := new(common.StatusCommunityLinkPreview)
+
+	// First, fill the output with the data from URL
+	c.CommunityID = data.CommunityID
+	c.DisplayName = data.DisplayName
+	c.Description = data.Description
+	c.MembersCount = data.MembersCount
+	c.Color = data.Color
+	c.TagIndices = data.TagIndices
+
+	// Now check if there's newer information in the database
+	communityID, err := types.DecodeHex(data.CommunityID)
+	if err != nil {
+		return c, fmt.Errorf("failed to decode community id: %w", err)
+	}
+
+	community, err := u.m.GetCommunityByID(communityID)
+	if err != nil {
+		return c, nil
```

Comment:
shouldn't we return err?

Reply:
Here I think we shouldn't, but we probably should above, if `types.DecodeHex` failed 🙂